### PR TITLE
crate v0

### PR DESCRIPTION
**Pull request by justinzhou93**

### CODE REVIEW

**Comment by justinzhou93** [comment]
2860 lines are cargo lock. so it's not that bad despite how the number of lines looks

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+[package]
+name = "oak_xcm"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (rhuttman) [inline Cargo.toml]: remove

File: src/xcm_config.rs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+#[derive(Encode, Decode, RuntimeDebug)]
+pub enum NeuChainCall<T: frame_system::Config> {
+    #[codec(index = 0)]
+    System(SystemCall),
+    #[codec(index = 60)]
+    AutomationTime(AutomationTimeCall<T>),
+}
+
+#[derive(Encode, Decode, RuntimeDebug)]
+pub enum TestChainCall {
```
Comment (rhuttman) [inline src/xcm_config.rs]: this should not live in the crate

File: src/xcm_config.rs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    #[codec(index = 8)]
+    RemarkWithEvent(Vec<u8>),
+}
+
+#[derive(Encode, Decode, RuntimeDebug)]
+pub enum AutomationTimeCall<T: frame_system::Config> {
+    #[codec(index = 0)]
+    ScheduleNotifyTask(Vec<u8>, Vec<u64>, Vec<u8>),
+    #[codec(index = 2)]
+    ScheduleXcmpTask(Vec<u8>, Vec<u64>, ParaId, Vec<u8>, u64),
```
Comment (rhuttman) [inline src/xcm_config.rs:12]: only this extrinsic is setup to accept `OriginKind::Native`. All the others will fail due to a bad origin. We either need to send those with the sovereign origin type or not support them in the crate. I vote for not supporting them as they add no real value.

File: src/xcm_config.rs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    #[codec(index = 3)]
+    ScheduleNativeTransferTask(Vec<u8>, Vec<u64>, T::AccountId, u128),
+    #[codec(index = 4)]
+    ScheduleCancelTask(T::Hash),
```
Comment (rhuttman) [inline src/xcm_config.rs]: based on the above will need to update the cancel to accept the new origin type, but I don't think we need that in this PR

File: src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+        execution_times: Vec<u64>,
+        para_id: ParaId,
+        returnable_call: Vec<u8>,
+    ) -> xcm::v2::Instruction<()> {
+        let call = xcm_config::OakChainCallBuilder::automation_time_schedule_xcmp::<T>(
+            provided_id,
+            execution_times,
+            para_id,
+            returnable_call,
+            6_000_000_000,
```
Comment (rhuttman) [inline src/lib.rs]: where is this number coming from? you at least need a comment to go w/ the magic number

**Review by rhuttman — CHANGES_REQUESTED** [review]
Will need to add a PR with doc comments and a README

File: src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,109 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+/// Edit this file to define custom logic or remove it if it is not needed.
```
Comment (rhuttman) [inline src/lib.rs]: remove

File: src/xcm_config.rs [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+#[derive(Encode, Decode, RuntimeDebug)]
+pub enum AutomationTimeCall {
+    #[codec(index = 2)]
+    ScheduleXcmpTask(Vec<u8>, Vec<u64>, ParaId, Vec<u8>, u64),
+}
+
+#[derive(Encode, Decode, RuntimeDebug)]
+pub enum NeuChainCall {
+    #[codec(index = 0)]
+    System(SystemCall),
```
Comment (rhuttman) [inline src/xcm_config.rs]: remove

File: src/xcm_config.rs [hunk 8]
```diff
@@ -0,0 +1,47 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+
+use cumulus_primitives_core::ParaId;
+use frame_support::RuntimeDebug;
+use sp_std::prelude::*;
+
+#[derive(Encode, Decode, RuntimeDebug)]
+pub enum SystemCall {
```
Comment (rhuttman) [inline src/xcm_config.rs]: remove

**Review by rhuttman — CHANGES_REQUESTED** [review]
a few missed bits of dead code

**Review by rhuttman — APPROVED** [review]

**Pull request merged**